### PR TITLE
Main Screen View Model Tests Written

### PR DESCRIPTION
**Pull request by Manik2708**
**What kind of change does this PR introduce?**
1) Improved the code coverage of the `MainScreenViewModel.dart`
2) Added multiple tests to improve the effectiveness of tests
3) Some of tests were not matching/expecting anything, improved those tests.

**Issue Number:**

Fixes #1121

**Did you add tests for your changes?**

Yes

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

File: test/view_model_tests/main_screen_view_model_test.dart [hunk 1]
```diff
@@ -327,139 +368,380 @@ void main() async {
...
+        (tester) async {
+      final int randomNumber = Random().nextInt(10);
+      final GlobalKey<ScaffoldState> key = MainScreenViewModel.scaffoldKey;
+      final model = getAndRegisterUserConfig();
+      const val1 = false;
+      when(model.loggedIn).thenAnswer((_) => val1);
+      final List<FocusTarget> expectedTargets = [];
+      late Map<String, GlobalKey> keysMap;
+      // locator.registerFactory(() => CustomDrawerViewModel());
+      final app = MaterialApp(
```
[the remark this answers is not in the document]
Comment (Manik2708) [reply test/view_model_tests/main_screen_view_model_test.dart]: I have made the changes. Please Review!

File: lib/view_model/main_screen_view_model.dart [hunk 2]
```diff
@@ -396,7 +397,8 @@ class MainScreenViewModel extends BaseModel {
   ///
   /// **returns**:
   ///   None
-  void tourHomeTargets() {
+  void tourHomeTargets([UserConfig? givenUserConfig]) {
```
[the remark this answers is not in the document]
Comment (literalEval) [reply lib/view_model/main_screen_view_model.dart:400]: I understand the rationale, but mocking will be a better way to go. Please look into mocking with mockito.

File: test/view_model_tests/main_screen_view_model_test.dart [hunk 3]
```diff
@@ -48,6 +48,31 @@ class MockLocalMainScreenViewModel extends MainScreenViewModel {
   }
 }
 
+typedef FunctionType = void Function(MainScreenViewModel model2);
+Widget getMockApp(FunctionType onTap) {
```
Comment (literalEval) [inline test/view_model_tests/main_screen_view_model_test.dart:52]: 1. Add a new line here.
2. Rename the function to be consistent with the naming used in other tests.

**Review by literalEval — CHANGES_REQUESTED** [review]

File: lib/view_model/main_screen_view_model.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (Manik2708) [reply lib/view_model/main_screen_view_model.dart:400]: > I understand the rationale, but mocking will be a better way to go. Please look into mocking with mockito.

So is there a way, to achieve this without injecting the mock model? I looked on other tests but found nothing to control the behaviour of UserConfig model when used inside a function. I am always doubtful of injecting dependency in this way, so would be very thankful if you can assist me!

**Comment by noman2002** [comment]
@Manik2708 Any update on this?

**Comment by Manik2708** [comment]
> @Manik2708 Any update on this?

There was some medical emergency at my home. Will update the PR till Monday.

**Comment by palisadoes** [comment]
I hope all is well with you and your family that the recovery is speedy

**Comment by Manik2708** [comment]
> I hope all is well with you and your family that the recovery is speedy

All is fine now! Thanks a lot for your wishes @palisadoes.

**Comment by palisadoes** [comment]
1. Why was this closed? I've noticed that there is a trend to do this recently without explanation.
2. It is a real burden for us to keep assigning reviewers only to have their work discarded with this type of action
3. Please be respectful of our developers time

**Comment by Manik2708** [comment]
> 1. Why was this closed? I've noticed that there is a trend to do this recently without explanation.
> 2. It is a real burden for us to keep assigning reviewers only to have their work discarded with this type of action
> 3. Please be respectful of our developers time

I am so sorry for this, actually I updated the PR after installation of some dependencies leading to high number of conflicts. I will keep in mind to state reason next time before closing and before that I will try not to close it.

**Pull request closed without merging**